app/views/planning_applications/show.html.erb

```diff
@@ -1,5 +1,101 @@
+                            aria: { describedby: "validation-completed" }
+                %>
+              <% else %>
+                Validate documents
+              <% end %>
+            </span>
+
+            <% if @planning_application.validation_complete?  %>
+              <%= tag.strong "Completed", id: "validation-completed", class: "govuk-tag app-task-list__task-tag" %>
+            <% end %>
+          </li>
+
+          <li class="app-task-list__item">
+            <span class="app-task-list__task-name">
+              <% if @planning_application.can_assess? %>
+                <%= link_to "Assess Proposal",
+                            recommendation_form_planning_application_path(@planning_application),
+                            aria: { describedby: "assessment-completed" }
+                %>
+              <% else %>
+                Assess Proposal
+              <% end %>
+            </span>
+
+            <% if @planning_application.assessment_complete?  %>
+              <%= tag.strong "Completed", id: "validation-completed", class: "govuk-tag app-task-list__task-tag" %>
+            <% end %>
+          </li>
+
+          <li class="app-task-list__item">
+            <span class="app-task-list__task-name">
+              <% if @planning_application.can_submit_recommendation? %>
+                <%= link_to "Submit Recommendation",
+                            submit_recommendation_planning_application_path(@planning_application),
+                            aria: { describedby: "submit_recommendation-completed" }
+                %>
+              <% else %>
+                Submit Recommendation
+              <% end %>
+            </span>
+
+            <% if @planning_application.submit_recommendation_complete?  %>
+              <%= tag.strong "Completed", id: "submit_recommendation-completed", class: "govuk-tag app-task-list__task-tag" %>
+            <% end %>
+          </li>
+
+          <li class="app-task-list__item">
+            <span class="app-task-list__task-name">
+              <% if @planning_application.can_review_assessment? && current_user.reviewer? %>
+                <%= link_to "Review Assessment",
+                            review_form_planning_application_path(@planning_application),
+                            aria: { describedby: "review_assessment-completed" }
+                %>
+              <% else %>
+                Review Assessment
+              <% end %>
+            </span>
+
+            <% if @planning_application.review_assessment_complete? %>
+              <%= tag.strong "Completed", id: "review_assessment-completed", class: "govuk-tag app-task-list__task-tag" %>
+            <% elsif @planning_application.can_review_assessment? && current_user.assessor? %>
+              <%= tag.strong "Waiting", id: "review_assessment-waiting", class: "govuk-tag app-task-list__task-tag" %>
+            <% end %>
+          </li>
+
+          <li class="app-task-list__item">
```

Comment:
I'm not sure this flow is quite right now I'm looking at it a second time (we can always fix later as a chore, so it doesn't have to be a blocker. We are removing this link once the application has been determined. This is kind of correct behaviour because obviously someone should not be able to publish the application twice. But it means that the application cannot be viewed once it has been determined. Probably a question for Product.